packages/typescript-plugin/lib/requests/getImportPathForFile.ts

```diff
@@ -1,14 +1,66 @@
 import type * as ts from 'typescript';
 
+const enum ExportKind {
+	Named,
+	Default,
+	ExportEquals,
+	UMD,
+}
+
+interface SymbolExportInfo {
+	readonly symbol: Symbol;
+	readonly moduleSymbol: Symbol;
+	/** Set if `moduleSymbol` is an external module, not an ambient module */
+	moduleFileName: string | undefined;
+	exportKind: ExportKind;
+	targetFlags: ts.SymbolFlags;
+	/** True if export was only found via the package.json AutoImportProvider (for telemetry). */
+	isFromPackageJson: boolean;
+}
+
 export function getImportPathForFile(
 	this: {
 		typescript: typeof import('typescript');
 		languageService: ts.LanguageService;
+		languageServiceHost: ts.LanguageServiceHost;
 	},
 	fileName: string,
-	importFileName: string,
+	incomingFileName: string,
+	preferences: ts.UserPreferences,
 ) {
-	const { typescript: ts, languageService } = this;
-	// TODO
-	return importFileName.replace('./', '@/');
+	const { typescript: ts, languageService, languageServiceHost } = this;
+	const { createImportSpecifierResolver }: {
+		createImportSpecifierResolver(
+			importingFile: ts.SourceFile,
+			program: ts.Program,
+			host: ts.LanguageServiceHost,
+			preferences: ts.UserPreferences
+		): {
+			getModuleSpecifierForBestExportInfo(
+				exportInfo: readonly SymbolExportInfo[],
+				symbolName: string,
+				position: number,
+				isValidTypeOnlyUseSite: boolean,
+				fromCacheOnly?: boolean
+			): { exportInfo?: SymbolExportInfo, moduleSpecifier: string, computedWithoutCacheCount: number; } | undefined;
+		};
+	} = (ts as any).codefix;
```

Comment:
How does it work? 🧐
Will `typescript` secretly export some internal services?

Reply:
Yes, some APIs are not exposed in dts.